src/Servers/HttpSys/test/FunctionalTests/Listener/RequestTests.cs

```diff
@@ -142,6 +136,7 @@ public async Task Request_OverlongUTF8Path(string requestPath, string expectedPa
     [InlineData("/", "/", "", "/")]
     [InlineData("/base", "/base", "/base", "")]
     [InlineData("/base", "/baSe", "/baSe", "")]
+    [InlineData("/base", "/baSe/", "/baSe", "/")]
```

Comment:
There was a branch in Request.ctor that wasn't being hit by tests, this test case hits it